lab-andrew/lib/client-pool.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const EventEmitter = require('events').EventEmitter;
+
+function random() {
+  return Math.floor(Math.random() * 5);
+}
+var nickNames = ['Troll', 'Mr. Smith', '42', 'Curry', 'Matt but better'];
+
+function registerClient(socket, clientPool) {
+  socket.wack = {};
+  socket.wack.id = nickNames[random()];
+  clientPool.pool[socket.wack.id] = socket;
```

Comment:
The user's id on the pool and nickname should be different.  Please add a nickname ("nick") property which is assigned a unique nickname.